examples/proposed/counting-semaphore.w

```diff
@@ -0,0 +1,75 @@
+bring cloud;
+
+struct CountingSemaphoreProps {
+  availableResources: num;
+}
+
+resource CountingSemaphore {
+  public limit: num;
+  _counter: cloud.Counter;
+
+  // need some ttl solution here,
+  // probably in-house once replaced with a key-value store
+  init(props: CountingSemaphoreProps) {
+    // pseudocode: input validation
+    this.limit = props.availableResources;
+    this._counter = new cloud.Counter();
+  }
+
+  // some stable unique instance id is wanted in the inflight context
+  // so that a reousce instance can be properly claimed and later released
+  // when used in conjunction with a key-value store
+  public inflight try_acquire(): bool {
```

Comment:
compiler seems to be not happy with `public inflight`

wondering if `public` modifier is necessary in this case? or recommended in general? as I saw some inconsistency across examples in the repo, please advice

Reply:
bug reported: https://github.com/winglang/wing/issues/1271